packages/lit-element/src/lib/updating-element.ts

```diff
@@ -789,17 +772,18 @@ export abstract class UpdatingElement extends HTMLElement {
    * Setting properties inside this method will *not* trigger
    * another update.
    *
-   * @param _changedProperties Map of changed properties with old values
-   */
-  protected update(_changedProperties: PropertyValues) {
-    if (this._reflectingProperties !== undefined) {
-      // Use forEach so this works even if for/of loops are compiled to for
-      // loops expecting arrays
-      this._reflectingProperties.forEach((v, k) =>
-        this._propertyToAttribute(k, this[k as keyof this], v)
-      );
-      this._reflectingProperties = undefined;
-    }
+   * @param changedProperties Map of changed properties with old values
+   */
+  protected update(changedProperties: PropertyValues) {
+    // Use forEach so this works even if for/of loops are compiled to for
+    // loops expecting arrays
+    changedProperties.forEach((_v, k) =>
```

Comment:
Why do we do the `options.reflect` test in `_propertyToAttribute` rather than before calling `_propertyToAttribute`?  This was surprising to me, seems like `_propertyToAttribute` should do the thing, given its name?  It also wastes an [`attributeNameForProperty`](https://github.com/Polymer/lit-html/blob/28be4b442d27b2a1cd3693113a1a5432555000c2/packages/lit-element/src/lib/updating-element.ts#L540) lookup for every changed property, regardless of whether it is reflecting now that we're looping over all changed properties, not just the reflecting ones.

Reply:
Done.